components/brave_ads/browser/BUILD.gn

```diff
@@ -42,26 +42,40 @@ source_set("browser") {
       "ads_service_impl.h",
       "background_helper.cc",
       "background_helper.h",
-      "background_helper_linux.cc",
-      "background_helper_linux.h",
-      "background_helper_mac.mm",
-      "background_helper_mac.h",
-      "background_helper_win.cc",
-      "background_helper_win.h",
       "locale_helper.cc",
       "locale_helper.h",
-      "locale_helper_linux.cc",
-      "locale_helper_linux.h",
-      "locale_helper_mac.mm",
-      "locale_helper_mac.h",
-      "locale_helper_win.cc",
-      "locale_helper_win.h",
-      "locale_helper_android.cc",
-      "locale_helper_android.h",
       "bundle_state_database.cc",
       "bundle_state_database.h",
     ]
 
+    if (is_android) {
+      sources += [
+        "background_helper_android.cc",
```

Comment:
these files should not be conditional as gn works out which files it should build depending on the environment